pandas/io/excel/_base.py

```diff
@@ -1064,21 +1066,12 @@ def __init__(
             )
 
         if engine is None:
-            if ext == "ods":
-                engine = "odf"
-            elif ext == "xls":
-                engine = "xlrd"
-            else:
-                # GH 35029 - Prefer openpyxl except for xls files
-                if (
-                    import_optional_dependency(
-                        "openpyxl", raise_on_missing=False, on_version="ignore"
-                    )
-                    is not None
-                ):
-                    engine = "openpyxl"
-                else:
-                    engine = "xlrd"
+            try:
+                engine = config.get_option(f"io.excel.{ext}.reader", silent=True)
```

Comment:
What is the error message if this is set to a nonexistent engine?

Reply:
I thought this would raise, but it doesn't apparently since pd.set_option doesn't validate options for some reason. I will add such validation into this PR.

Reply:
Fixed, it will be of the form "Value Error: Value must be one of" and the valid engines list.